### PR TITLE
add feature: generate one postman collection of all modules

### PR DESCRIPTION
**Pull request by Ahmed-Rafat**

### CODE REVIEW

File: src/Console/Commands/PostmanCollection.php [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+     * 
+     * @return void
+    */
+    private function loadPostmanFiles() 
+    {
+        $modulePath = $this->getModulePath();
+
+        $this->postmanFiles = array_filter(glob($modulePath . '/*/' . self::DOCS_FOLDER_NAME . '/*.postman.json'));
+
+        $this->info('Success Load Postman Files ..');
```
Comment (hassanzohdy) [inline src/Console/Commands/PostmanCollection.php]: Successfully Loaded Postman Files...

File: src/Console/Commands/PostmanCollection.php [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+    /**
+     * Docs Folder Name
+    */
+    const DOCS_FOLDER_NAME = 'docs';
+
+    /**
+     * default generate file name
+    */
+    const DEFAULT_GENERATE_FILE_NAME = 'project-api-collection';
```
Comment (hassanzohdy) [inline src/Console/Commands/PostmanCollection.php]: DEFAULT_GENERATED_FILE_NAME

File: src/Console/Commands/PostmanCollection.php [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+     * @var string
+     */
+    protected $signature = 'engez:postman {--fileName=}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Generate Root Collection ..';
```
Comment (hassanzohdy) [inline src/Console/Commands/PostmanCollection.php]: Write more descriptive description.

File: src/Console/Commands/PostmanCollection.php [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+            'item' => [],
+            'variable' => [],
+            'info' => [
+                'name' => $appName,
+                'schema' => 'https://schema.getpostman.com/json/collection/v2.1.0/collection.json'
+            ]
+        ];
+
+        foreach($this->filesContent as $content) {
+            // Remove Content of Globel Variable in each Collection
```
Comment (hassanzohdy) [inline src/Console/Commands/PostmanCollection.php:119]: Perfectly written, i wouldn't get it if you didn't write these two lines (Y)

File: src/Console/Commands/PostmanCollection.php [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+namespace HZ\Illuminate\Mongez\Console\Commands;
+
+use Illuminate\Console\Command;
+
+class PostmanCollection extends Command
+{
+    /**
+     * Path Folder of modules
+    */
+    const MODULE_PATH = 'app/Modules';
```
Comment (hassanzohdy) [inline src/Console/Commands/PostmanCollection.php]: MODULES_PATH

File: src/Console/Commands/PostmanCollection.php [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+    {
+        return base_path(self::MODULE_PATH);
+    }
+
+    /**
+     * Return Module Repo
+     * 
+     * @return mixed
+    */
+    public function getRepositry($repo)
```
Comment (hassanzohdy) [inline src/Console/Commands/PostmanCollection.php]: Why this method exists?, if you want to get a repository use `repo` function instead, or inject the RepositoryTrait.

**Review by hassanzohdy — CHANGES_REQUESTED** [review]

File: src/Console/Commands/PostmanCollection.php (hunk 4 above)
Comment (Ahmed-Rafat) [reply src/Console/Commands/PostmanCollection.php:119]: Thanks, it was written by me.

File: src/Console/Commands/PostmanCollection.php (hunk 6 above)
Comment (Ahmed-Rafat) [reply src/Console/Commands/PostmanCollection.php]: sorry, this function is not important right now. Removed.

File: src/Console/Commands/PostmanCollection.php [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+{
+    /**
+     * Path Folder of modules
+    */
+    const MODULES_PATH = 'app/Modules';
+
+    /**
+     * Docs Folder Name
+    */
+    const DOCS_FOLDER_NAME = 'docs';
```
Comment (hassanzohdy) [inline src/Console/Commands/PostmanCollection.php]: DOCS_DIRECTORY_NAME*

File: src/Console/Commands/PostmanCollection.php [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+     * @var string
+     */
+    protected $description = 'Generate one postman collection of all modules.';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
```
Comment (hassanzohdy) [inline src/Console/Commands/PostmanCollection.php]: Remove the constructor as it is not needed.

File: src/Console/Commands/PostmanCollection.php [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+     * Get Global Variables
+     * 
+     * @return array
+    */
+    private function getGlobalVariables()
+    {
+        $allGlobalVariables = [];
+
+        foreach(array_column($this->filesContent, 'variable') as $globalVariable) {
+            foreach($globalVariable as $variable){
```
Comment (hassanzohdy) [inline src/Console/Commands/PostmanCollection.php]: Add Space before the brace

File: src/Console/Commands/PostmanCollection.php [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+    /**
+     * Create Root Collection
+     * 
+     * @return void
+    */
+    private function createCollection()
+    {
+        $appName = env('APP_NAME', 'Postman Collection with Mongez');
+
+
```
Comment (hassanzohdy) [inline src/Console/Commands/PostmanCollection.php]: Remove this line.

File: src/Console/Commands/PostmanCollection.php [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+        foreach(array_column($this->filesContent, 'variable') as $globalVariable) {
+            foreach($globalVariable as $variable){
+                $allGlobalVariables[$variable['key']] = $variable;
+            }
+        }
+        
+        return $allGlobalVariables;
+    }
+
+
```
Comment (hassanzohdy) [inline src/Console/Commands/PostmanCollection.php]: Remove this line.

**Review by hassanzohdy — CHANGES_REQUESTED** [review]
Check the comments on the file.

**Pull request merged**